helper_test.go

```diff
@@ -357,6 +361,59 @@ func createPlannedRun(t *testing.T, client *Client, w *Workspace) (*Run, func())
 	return r, rCleanup
 }
 
+func createCostEstimationOrganization(t *testing.T, client *Client) (*Organization, func()) {
+	ctx := context.Background()
+	org, err := client.Organizations.Create(ctx, OrganizationCreateOptions{
+		Name:                  String(randomString(t)),
+		Email:                 String(fmt.Sprintf("%s@tfe.local", randomString(t))),
+		CostEstimationEnabled: Bool(true),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return org, func() {
+		if err := client.Organizations.Delete(ctx, org.Name); err != nil {
+			t.Errorf("Error destroying organization! WARNING: Dangling resources\n"+
+				"may exist! The full error is shown below.\n\n"+
+				"Organization: %s\nError: %s", org.Name, err)
+		}
+	}
+}
+
+func createCostEstimatedRun(t *testing.T, client *Client) (*Run, func()) {
+	org, oCleanup := createCostEstimationOrganization(t, client)
+	w, wCleanup := createWorkspace(t, client, org)
+	r, rCleanup := createRun(t, client, w)
+
+	var err error
+	ctx := context.Background()
+	for i := 0; ; i++ {
+		r, err = client.Runs.Read(ctx, r.ID)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		switch r.Status {
+		case RunPlanned, RunCostEstimated, RunPolicyChecked, RunPolicyOverride:
+			break
+		}
+
+		if i > 30 {
+			rCleanup()
+			t.Fatal("Timeout waiting for run to be planned")
+		}
+
+		time.Sleep(1 * time.Second)
+	}
+
+	return r, func() {
```

Comment:
Maybe can put this in a call to defer, or put the infinite loop above in a goroutine? Is there prior art you can look at? this looks like an off pattern to me